python/ray/data/read_api.py

```diff
@@ -2122,42 +2094,38 @@ def from_huggingface(
             )
 
     Args:
-        dataset: A `Hugging Face Datasets Dataset`_ or `DatasetDict`_.
-            :class:`~ray.data.IterableDataset` isn't supported.
+        dataset: A `Hugging Face Datasets Dataset`_.
+            ``IterableDataset`` and
+            `DatasetDict <https://huggingface.co/docs/datasets/package_reference/main_classes#datasets.DatasetDict/>`_
+            are not supported.
 
     Returns:
-        A :class:`~ray.data.Dataset` holding rows from the `Hugging Face Datasets Dataset`_,
-        or a dict of :class:`Datasets <ray.data.Dataset>` in case ``dataset`` is a `DatasetDict`_.
+        A :class:`~ray.data.Dataset` holding rows from the `Hugging Face Datasets Dataset`_.
     """  # noqa: E501
     import datasets
 
-    def convert(ds: "datasets.Dataset") -> Dataset:
+    if isinstance(dataset, datasets.Dataset):
         # To get the resulting Arrow table from a Hugging Face Dataset after
         # applying transformations (e.g. train_test_split(), shard(), select()),
         # we create a copy of the Arrow table, which applies the indices
         # mapping from the transformations.
-        hf_ds_arrow = ds.with_format("arrow")
+        hf_ds_arrow = dataset.with_format("arrow")
         ray_ds = from_arrow(hf_ds_arrow[:])
         return ray_ds
-
-    if isinstance(dataset, datasets.DatasetDict):
+    elif isinstance(dataset, datasets.DatasetDict):
         available_keys = list(dataset.keys())
-        logger.warning(
-            "You provided a Huggingface DatasetDict which contains multiple "
-            "datasets. The output of `from_huggingface` is a dictionary of Ray "
-            "Datasets. To convert just a single Huggingface Dataset to a "
+        logger.error(
```

Comment:
don't we want to raise an exception here?

Reply:
ah okay, got it. let me raise a ValueError here